### PR TITLE
Adds servlets that collect feedback from users

### PR DESCRIPTION
**Pull request by GrantComm**
These servlets are called when a user submits a feedback form. These will be used for emailing feedback to users

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    if (interviewExists(scheduledInterviewId)) {
+      if (isInterviewer(scheduledInterviewId, userId)) {
+        response.sendRedirect("/scheduled-interviews.html");
+        return;
+      }
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    } else {
+      response.sendError(404);
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java]: Nit: can you get this in the same format at SC_UNAUTHORIZED
Same for other file

File: src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import java.time.temporal.ChronoUnit;
+import java.time.ZoneId;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+
+// Servlet that gets the feedback for an interviewer
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java]: Nit: make the distinction clear between sender and receiver
somthing like "from an interviewee and to an interviewer"
Same for other file

File: src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  private boolean interviewExists(long scheduledInterviewId) {
+    return scheduledInterviewDao.get(scheduledInterviewId).isPresent();
+  }
+
+  private boolean isInterviewer(long scheduledInterviewId, String userId) {
+    ScheduledInterview scheduledInterview = scheduledInterviewDao.get(scheduledInterviewId).get();
+    if ((!scheduledInterview.interviewerId().equals(userId))) {
+      return false;
+    }
+    return true;
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java]: Nit: `return scheduledInterview.interviewerId().equals(userId)`
Same for other file

File: src/main/webapp/feedback.js [hunk 4]
```diff
@@ -37,3 +37,9 @@ function getScheduledInterviewId() {
 function getRole() {
   return new URLSearchParams(window.location.search).get('role'); 
 }
+
+// Sends the feedback forms.
+function doSubmit() {
+  document.feedbackForm.interviewId.value = getScheduledInterviewId();
+  return true;
```
Comment (apluscs) [inline src/main/webapp/feedback.js:44]: Q: why does this return true if it's not being called anywhere?

File: src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    String question2 = request.getParameter("question2");
+    String question3 = request.getParameter("question3");
+    String question4 = request.getParameter("question4");
+    String question5 = request.getParameter("question5");
+    String question6 = request.getParameter("question6");
+    String question7 = request.getParameter("question7");
+    String question8 = request.getParameter("question8");
+    String question9 = request.getParameter("question9");
+    String question10 = request.getParameter("question10");
+    String question11 = request.getParameter("question11");
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java]: Can you put these in a list instead? Feel free to make it base-1 if it's easier.
Same for other file.

**Review by apluscs — CHANGES_REQUESTED** [review]
Small things

File: src/main/webapp/feedback.js (hunk 4 above)
Comment (GrantComm) [reply src/main/webapp/feedback.js:44]: It is called when the form is submitted. I use this to include the scheduledInterview's id in the servlet request

File: src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java (hunk 3 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java]: Done

File: src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java (hunk 2 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java]: Done

File: src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java (hunk 1 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java]: Done

File: src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java (hunk 5 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java]: Done

File: src/main/webapp/feedback.js (hunk 4 above)
Comment (apluscs) [reply src/main/webapp/feedback.js:44]: Ok l see it being called but the return value not being used.

File: src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    answers.add(request.getParameter("question2"));
+    answers.add(request.getParameter("question3"));
+    answers.add(request.getParameter("question4"));
+    answers.add(request.getParameter("question5"));
+    answers.add(request.getParameter("question6"));
+    answers.add(request.getParameter("question7"));
+    answers.add(request.getParameter("question8"));
+    answers.add(request.getParameter("question9"));
+    answers.add(request.getParameter("question10"));
+    answers.add(request.getParameter("question11"));
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java:64]: You can just loop through all the questions you want and specify the param as "question" + i.toString() or smthg
Same for other file

**Review by apluscs — APPROVED** [review]
1 nit, 1 Q
but LGTM

**Comment by GrantComm** [comment]
Closing this  because I have the emails working which for these servlets, comments on this pr were incorporated in that pr.

**Pull request closed without merging**